sphinx_needs/functions/functions.py

```diff
@@ -93,18 +119,24 @@ def execute_func(app: Sphinx, need: NeedsInfoType, func_string: str) -> Any:
         **func_kwargs,
     )
 
-    if not isinstance(func_return, (str, int, float, list, unicode)) and func_return:
-        raise SphinxError(
-            f"Return value of function {func_name} is of type {type(func_return)}. Allowed are str, int, float"
+    if func_return is not None and not isinstance(func_return, (str, int, float, list)):
+        log_warning(
+            logger,
+            f"Return value of function {func_name!r} is of type {type(func_return)}. Allowed are str, int, float, list",
+            "dynamic_function",
+            location=location,
         )
-
+        return "??"
     if isinstance(func_return, list):
         for element in func_return:
-            if not isinstance(element, (str, int, float, unicode)):
-                raise SphinxError(
-                    f"Element of return list of function {func_name} is of type {type(func_return)}. "
-                    "Allowed are str, int, float"
+            if not isinstance(element, (str, int, float)):
+                log_warning(
+                    logger,
+                    f"Return value of function {func_name!r} is of type {type(func_return)}. Allowed are str, int, float, list",
```

Comment:
Can you please include the list index to the message? The type is already a list, just one value has a wrong type.